optimizationmodule/opt_module_test.go

```diff
@@ -0,0 +1,26 @@
+package optimizationmodule
+
+import (
+	"encoding/json"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestExecuteRules(t *testing.T) {
+
+	rules := Build(json.RawMessage{})
+	rw := &openrtb_ext.RequestWrapper{
+		BidRequest: &openrtb2.BidRequest{
+			Device: &openrtb2.Device{
+				Geo: &openrtb2.Geo{
+					Country: "USA",
+				},
+				IP: "0.0.0.0",
+			},
+		}}
+	err := rules.Execute(rw)
```

Comment:
The params first need to be extracted, then given to Execute, and then Execute will use the params to locate a node. This will involve some backtracking behavior for defaults.